hack/prow_test_dumper.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+"""
+Dump required tests out of prow config in human (and machine) readable format
+"""
+
+from argparse import Namespace, ArgumentParser
+from typing import Dict
+
+# pip3 install PyYAML
+import yaml
+
+
+def get_external_branchprotection(data: dict) -> Dict[str, list]:
+    """print external branch protection jobs"""
+    protected_repos: Dict[str, list] = {}
+
+    for org, org_data in data.get("orgs", {}).items():
+        for repo, repo_data in org_data.get("repos", {}).items():
+            for branch, branch_data in repo_data.get("branches", {}).items():
+                required_checks = branch_data.get("required_status_checks", {}).get("contexts", [])
+                key = f"{org}/{repo}/{branch}"
+                if key not in protected_repos:
+                    protected_repos[key] = required_checks
+                else:
+                    protected_repos[key].append(required_checks)
+
+    return protected_repos
+
+
+def get_prow_branchprotection(data: dict) -> Dict[str, list]:
+    """get prow required tests"""
+    protected_repos: Dict[str, list] = {}
```

Comment:
Same here

Reply:
Cleaned this up in many ways, including defaultdict. Thanks.